Source/santa-driver/SantaDecisionManager.cc

```diff
@@ -208,6 +219,83 @@ kern_return_t SantaDecisionManager::StopListener() {
   return kIOReturnSuccess;
 }
 
+# pragma mark Monitoring PIDs
+
+// Arguments that are passed to pid_monitor thread.
+typedef struct {
+  pid_t pid;                  // process to monitor
+  SantaDecisionManager *sdm;  // reference to SantaDecisionManager
+} pid_monitor_info;
+
+// Function executed in its own thread used to monitor a compiler process for
+// termination and then remove the process pid from cache of compiler pids.
+static void pid_monitor(void *param, __unused wait_result_t wait_result) {
+  pid_monitor_info *info = (pid_monitor_info *)param;
+  uint32_t sleep_time = info->sdm->PidMonitorSleepTimeMilliseconds();
+  if (info->sdm) {
```

Comment:
To be on the safe side I'd check info is not nullptr first too

Reply:
Done.